mobile/voice_journal_app/lib/RecordingPage.dart

```diff
@@ -67,27 +86,36 @@ pauseRecording(){
   message = 'Recording paused';
   displaySaved();
 }
+void main() async{
+  runApp(const MaterialApp(home: HomePage())); //Run the homepage if this page is ran on its own
+}
 class RecordingPage extends StatefulWidget{
-  const RecordingPage({super.key, required this.title});
+  const RecordingPage({super.key, required this.title, required this.callback}); //takes in call back function to call when this page is closed, this refreashes the widget tree on the homepage to update the recent recordings list.
   final String title;
+  final VoidCallback callback;
   @override
   State<RecordingPage> createState() => _RecordingPageState();
 }
 class _RecordingPageState extends State<RecordingPage>{
   void _changeState() async{
-    if(!recording){ //if currently not recording and the button has been hit`
-      recording = true; //recording is true
-      await startRecording(); //wait for function to start the recording
-      if(!counting){
-        _startCounting(); //Starts the displayed recording counter
+    final hasPermission = await controller.checkPermission(); //get permission to use mic
+    if(hasPermission){
+      if(!recording){ //if currently not recording and the button has been hit`
+        //recording = true; //recording is true
+        await startRecording(); //wait for function to start the recording
+        if(!counting){
+          _startCounting(); //Starts the displayed recording counter
+        }
+        setState(() { //change state (icon of the button should change)
+        });
```

Comment:
instead of setting empty states, let's set `recording = true` for the new state instead of doing it in the `startRecording` function. Any variables used to render widgets in the `build` function should be set in the state.